api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -273,8 +273,10 @@
 
     /**
      * Return all of the currently registered {@linkplain ConfigSource configuration sources} for this configuration.
+     * <p>
      * The returned sources will be sorted by descending ordinal value and name, which can be iterated in a thread-safe manner.
-     *
+     * The {@link java.util.Iterator Iterable} is not alterable and 
+     * contains a fixed number of {@linkplain ConfigSource configuration sources}, which may be either static or dynamic.
```

Comment:
I find the wording very vague and easy to misunderstand. 

Ideas you could (wrongly) get:
* _not alterable_ means you will get errors when trying the `remove()` method?
* _not alterable_ means means every call might returns the very same `Iterable` instance
* _which may be either static or dynamic_ means the iterated elements (the length and order of the backing collection) may or may not change

Ideas you might not get but should:
* every call returns a new `Iterable` instance but the elements iterated never change (when are they pinned down? bootstrapping, point of first use?)

Reply:
Good point @jbee! I tried a few variants, and then settle on the current statement. I think what you proposed is better. I will update accordingly.

Reply:
@jbee I thought it further and altered the above comments. Can you double check?

Reply:
> The Iterable contains a fixed number of config sources, determined at application start time, and the config sources may be static or dynamic.

More clear. Go for it.

Reply:
Well, now, wait a minute: can't a `ConfigSourceProvider` supply a set of `ConfigSources`?  So if a `Config` uses that set for its own `ConfigSource`s, then the set's "staticness" cannot be guaranteed, right?  Nor, presumably, can the iteration behavior be guaranteed?  Is it written somewhere that a `ConfigSourceProvider` must provide a snapshot?

Reply:
Yep; just checked the [`ConfigSourceProvider` javadocs](https://javadoc.io/static/org.eclipse.microprofile.config/microprofile-config-api/1.4/org/eclipse/microprofile/config/spi/ConfigSourceProvider.html) and there is nothing specified about how this must or must not behave. Nor is there anything specified about how `Config` must represent its output. So that may need to change before we make a guarantee here.

Reply:
Hum, I believe that we need to change other parts of the specification if we want to make this change.

From the spec:

> MicroProfile Config also allows to implement and register own configuration sources in a portable way, e.g. for reading configuration values from a shared database in an application cluster.

From my understanding, there might be non-portable ways to register Config Sources, including after the `Config` object is initialized. In that particular case, then `getConfigSources` may be dynamic as well. If it needs to be determined at `Config` startup, then the spec must also change (or provide clarification), to Config Sources can only be registered / discovered on `Config` init.

Reply:
I have made the similar javadocs on `ConfigSourceProvider` @ljnelson. Please check to see whether you have any further comments.